### PR TITLE
adds remove tenant functionality, by deleting the lease, and fetching…

### PR DESCRIPTION
**Pull request by Protopaco**
… property data

### What issue is this solving?
Allows user to delete a tenant's lease, thus removing them from the property.  This change only removes the lease, it does not remove the tenant.

It also removes the "getTenants()" function as that information now comes in with property object.

### Any helpful knowledge/context for the reviewer?
Is a `npm install` necessary? no
Any special requirements to test? no

### Please make sure you've attempted to meet the following coding standards
- [ ] Code builds successfully
- [ ] Code has been tested and does not produce errors
- [ ] Code is readable and formatted
- [ ] There isn't any unnecessary commented-out code
- [ ] There aren't any unnecessary commits or files changed (shouldn't touch the /stashed folder unless the issue requests it)

If you're having trouble meeting this criteria, feel free to reach out on Slack for help!

### CODE REVIEW

File: src/views/Property/PropertyView.jsx [hunk 1]
```diff
@@ -201,9 +189,20 @@ const Property = () => {
...
 
-    getProperty()
+  const handleTenantConfirmButton = () => {
+    const leaseId = tenantToRemove.lease.id;
+    axios
+      .delete(`${process.env.REACT_APP_PROXY}/api/lease/${leaseId}`)
+      .then(res => getProperty())
+      .then(setConfirmTenantRemoval(false))
+      .then(setEditingStatus(false))
+      .then(Toast("Tenant successfully removed"))
```
Comment (DaveTrost) [inline src/views/Property/PropertyView.jsx:204]: Once the response is received (line 201), further calls are not dependent on the completion of those prior. In other words, this section needs only one ".then".

**Review by DaveTrost — CHANGES_REQUESTED** [review]
After removing a tenant from the property, the tenant's detail page gives the following error: "Unhandled Rejection (TypeError): Cannot read property 'address' of undefined". I think a tenant without a property needs to be handled cleanly before merging this PR.

The remove button looks out of place in this app. Any ideas for giving it a look more consistent with the site design are welcome!

Something bothers me about the way edit mode works with the tenant removal feature. Probably fine for now, but I have ideas for the long run. I think removing tenants from the property is best done without needing to enter edit mode for the property. This will essentially separate property modifications (address, name, manager) from lease modifications (add/remove tenants).

File: src/views/Property/PropertyView.jsx (hunk 1 above)
Comment (Protopaco) [reply src/views/Property/PropertyView.jsx:204]: This has been corrected.  Thank you!

**Comment by Protopaco** [comment]
> After removing a tenant from the property, the tenant's detail page gives the following error: "Unhandled Rejection (TypeError): Cannot read property 'address' of undefined". I think a tenant without a property needs to be handled cleanly before merging this PR.
> 
> The remove button looks out of place in this app. Any ideas for giving it a look more consistent with the site design are welcome!
> 
> Something bothers me about the way edit mode works with the tenant removal feature. Probably fine for now, but I have ideas for the long run. I think removing tenants from the property is best done without needing to enter edit mode for the property. This will essentially separate property modifications (address, name, manager) from lease modifications (add/remove tenants).

I agree about how TenantView handles Tenants without Leases.  It seems like it could with some conditional rendering.  Is that best handled through writing a new ticket?

I updated the remove tenant icon to make it more consistent with the app.

I agree about how the edit mode works.  It seems like editing the Property attributes, Property Managers, and Tenants could all be handled separately.  For this I tried to follow the FIGMA's layout, but I would be happy to break them out.  

Thanks!

**Comment by DaveTrost** [comment]
> I updated the remove tenant icon to make it more consistent with the app.
> 

Do you know why the cursor effect on removing a property manager does not change to a pointer?

**Comment by DaveTrost** [comment]
> I agree about how TenantView handles Tenants without Leases. It seems like it could with some conditional rendering. Is that best handled through writing a new ticket?

Yes it is! We talked about this in the development meeting and decided to open a separate ticket: https://github.com/codeforpdx/dwellingly-app/issues/534

File: src/views/Property/PropertyView.jsx (hunk 1 above)
Comment (Chris-Boe) [reply src/views/Property/PropertyView.jsx:204]: I'm not seeing this change

**Review by Chris-Boe — CHANGES_REQUESTED** [review]
I think this just needs to be rebased and get merged conflicts resolved, as well as the change Dave requested done. I'm having trouble testing it again with all the updates both the frontend and backend have had

**Pull request closed without merging**